app/services/atmosphere/optimizer.rb

```diff
@@ -14,6 +14,10 @@ def run(hint)
 
     #private
     def satisfy_appliance(appliance)
+      action = Action.create(appliance: appliance,
```

Comment:
Everything is righ @paoolo. At this place satisfy_appliance represents whole new action so a new Action is created. In the body of this method you can call log several times on that object.